src/Components/Search/NewSearch/NewSearchBarInput.tsx

```diff
@@ -10,92 +9,234 @@ import { NewSearchBarInput_viewer$data } from "__generated__/NewSearchBarInput_v
+  // We use fetchCounter together with useUpdateEffect to track typing
+  const [fetchCounter, setFetchCounter] = useState(0)
+  const { router } = useRouter()
+  const encodedSearchURL = `/search?term=${encodeURIComponent(value)}`
 
   const options = extractNodes(viewer.searchConnection)
-  const formattedOptions = options.map(option => {
-    return {
-      text: option.displayLabel!,
-      value: option.displayLabel!,
-      subtitle: getLabel({
-        displayType: option.displayType ?? "",
+  const formattedOptions: SuggionItemOptionProps[] = options.map(
+    (option, index) => {
+      return {
+        text: option.displayLabel!,
+        value: option.displayLabel!,
+        subtitle:
+          getLabel({
+            displayType: option.displayType ?? "",
+            typename: option.__typename,
+          }) ?? "",
+        imageUrl: option.imageUrl!,
+        showArtworksButton: !!option.statuses?.artworks,
+        showAuctionResultsButton: !!option.statuses?.auctionLots,
+        href: option.href!,
         typename: option.__typename,
-      }),
-      imageUrl: option.imageUrl!,
-      showArtworksButton: !!option.statuses?.artworks,
-      showAuctionResultsButton: !!option.statuses?.auctionLots,
-      href: option.href!,
+        item_number: index,
+        item_type: option.displayType!,
+      }
     }
-  })
+  )
 
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setValue(event.target.value)
+  // Clear the search term once you navigate away from search results
+  useMemo(() => {
+    router.addNavigationListener(location => {
+      if (!location.pathname.startsWith("/search")) {
+        setValue("")
+      }
+
+      return true
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useUpdateEffect(() => {
+    tracking.trackEvent({
+      action_type:
+        options.length > 0
+          ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
+          : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
+      query: value,
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [fetchCounter])
+
+  const refetch = (value: string, entity?: string) => {
+    const entities = entity ? [entity] : []
+    const performanceStart = performance && performance.now()
 
-    //TODO: Add throttle
     relay.refetch(
       {
         hasTerm: true,
-        term: event.target.value,
+        term: value,
+        entities: entities,
       },
       null,
       error => {
         if (error) {
           logger.error(error)
           return
         }
+
+        if (performanceStart && getENV("VOLLEY_ENDPOINT")) {
+          reportPerformanceMeasurement(performanceStart)
+        }
+
+        setFetchCounter(prevCounter => prevCounter + 1)
       }
     )
   }
 
+  const debouncedSearchRequest = useMemo(
+    () => debounce(refetch, SEARCH_DEBOUNCE_DELAY),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
So debouncing in React hooks is non-trivial. This is papering over a bug. I'd just import the `use-debounce` package